go/libkb/socket_windows.go

```diff
@@ -9,43 +9,36 @@ package libkb
 import (
 	"errors"
 	"net"
-	"os/user"
+	"path/filepath"
+	"strings"
 
 	"github.com/natefinch/npipe"
 )
 
-type SocketNamedPipe struct {
-	Contextified
-	pipename string
-}
-
-// user.Current() includes machine name on windows, but
-// this is still only a local pipe because of the dot
-// following the doulble backslashes.
-// If the service ever runs under a different account than
-// current user, this will have to be revisited.
 func NewSocket(g *GlobalContext) (ret Socket, err error) {
-	currentUser, err := user.Current()
+	var s string
+	s, err = g.Env.GetSocketFile()
 	if err != nil {
 		return
 	}
-	if len(currentUser.Username) == 0 {
-		err = errors.New("Empty username, can't make pipe")
+	if len(s) == 0 {
+		err = errors.New("Empty SocketFile, can't make pipe")
 		return
 	}
-	return SocketNamedPipe{
+	s = strings.TrimPrefix(s, filepath.VolumeName(s))
+	return SocketInfo{
 		Contextified: NewContextified(g),
-		pipename:     `\\.\pipe\kbservice\` + currentUser.Username,
+		file:         `\\.\pipe\kbservice` + s,
```

Comment:
what does the pipe wind up looking like?  will it have a mix of back and forward slashes?